pkg/cluster/cluster.go

```diff
@@ -0,0 +1,293 @@
+			}
+			return errors.New("aborting due to invalid configuration")
+		}
+		return errors.Wrap(err, "failed to create cluster")
+	}
+	wg.Done()
+	return nil
+}
+
+// Destroy destroys a kind cluster
+func Destroy(clName string, provider *kind.Provider) error {
+	log.Infof("Deleting cluster %q ...\n", clName)
+	if err := provider.Delete(clName, ""); err != nil {
+		return errors.Wrapf(err, "failed to delete cluster %s", clName)
+	}
+
+	usr, err := user.Current()
+	if err != nil {
+		return err
+	}
+
+	_ = os.Remove(filepath.Join(defaults.KindConfigDir, "kind-config-"+clName+".yaml"))
+	_ = os.Remove(filepath.Join(defaults.LocalKubeConfigDir, "kind-config-"+clName))
+	_ = os.Remove(filepath.Join(defaults.ContainerKubeConfigDir, "kind-config-"+clName))
+	_ = os.RemoveAll(filepath.Join(usr.HomeDir, ".kube", strings.Join([]string{"kind-config", clName}, "-")))
+	_ = os.RemoveAll(filepath.Join(defaults.KindLogsDir, clName))
+
+	return nil
+}
+
+// GetMasterDockerIP gets control plain master docker internal ip
+func GetMasterDockerIP(clName string) (string, error) {
+	ctx := context.Background()
+	dockerCli, err := dockerclient.NewEnvClient()
+	if err != nil {
+		return "", err
+	}
+
+	containerFilter := filters.NewArgs()
+	containerFilter.Add("name", strings.Join([]string{clName, "control-plane"}, "-"))
+	containers, err := dockerCli.ContainerList(ctx, dockertypes.ContainerListOptions{
+		Filters: containerFilter,
+		Limit:   1,
+	})
+	if err != nil {
+		return "", err
+	}
+	return containers[0].NetworkSettings.Networks["bridge"].IPAddress, nil
+}
+
+// iterate func map for config template
+func iterate(start, end int) (stream chan int) {
+	stream = make(chan int)
+	go func() {
+		for i := start; i <= end; i++ {
+			stream <- i
+		}
+		close(stream)
+	}()
+	return
+}
+
+// IsKnown returns bool if cluster exists
+func IsKnown(clName string, provider *kind.Provider) (bool, error) {
+	n, err := provider.ListNodes(clName)
+	if err != nil {
+		return false, err
+	}
+	if len(n) != 0 {
+		return true, nil
+	}
+	return false, nil
+}
+
+// GetClientSet returns kubernetes interface
+func GetClientSet(clName string) (kubernetes.Interface, error) {
+	kubeConfigFilePath, err := GetKubeConfigPath(clName)
+	if err != nil {
+		return nil, err
+	}
+
+	kconfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	clientSet, err := kubernetes.NewForConfig(kconfig)
+	if err != nil {
+		return nil, err
+	}
+	return clientSet, nil
+}
+
+// GetCrdClientSet returns apiextclientset interface
+func GetCrdClientSet(clName string) (apiextclientset.Interface, error) {
+	kubeConfigFilePath, err := GetKubeConfigPath(clName)
+	if err != nil {
+		return nil, err
+	}
+
+	kconfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	apiExtClientSet, err := apiextclientset.NewForConfig(kconfig)
+	if err != nil {
+		return nil, err
+	}
+	return apiExtClientSet, nil
+}
```

Comment:
Practically the same code for both functions, I think you could extract commonalities and make them just wrappers to a clientset getter.

Reply:
`GetCrdClientSet` has been removed